includes/class-sensei-settings.php

```diff
@@ -1096,6 +1097,46 @@ public function experimental_features_saved( $old_value, $value ) {
 		}
 	}
 
+	/**
+	 * Renders the High-Performance Progress Storage feature setting.
+	 *
+	 * @param array $args The field arguments.
+	 */
+	public function render_progress_storage_feature( $args ) {
+
+		// Checkbox field.
+		$settings = $this->get_settings();
+		$key      = $args['key'];
+		$value    = $settings[ $key ];
+
+		// Disables the checkbox if we can't set time limit. That's our current limitation for running migrations.
+		$disabled  = ! set_time_limit( 0 );
+		?>
+		<div>
+			<label>
+				<input
+					class="sensei-settings_progress-storage-feature"
+					type="checkbox"
+					name="<?php echo esc_attr( "{$this->token}[{$key}]" ); ?>"
+					value="1"
+					<?php disabled( true, $disabled, true ); ?>
+					<?php checked( $value, true, true ); ?>
+				/>
+				<?php echo esc_html( $args['data']['description'] ); ?>
+			</label>
+
+			<?php if ( $disabled ) : ?>
+				<input type="hidden" name="<?php echo esc_attr( "{$this->token}[{$key}]" ); ?>" value="<?php echo esc_attr( $value ); ?>" />
+				<p>
+					<?php echo esc_html( __( 'Currently, to make the feature work, we need to execute long-running migrations.', 'sensei-lms' ) ) ?><br />
```

Comment:
For setting a time limit, presumably Woo had a similar issue. Can we research / ask Woo folks about how they handled it?